db/view/view.cc

```diff
@@ -1870,8 +1910,13 @@ future<> view_update_generator::mutate_MV(
         // TODO: Maybe allow users to set use_legacy_self_pairing explicitly
         // on a view, like we have the synchronous_updates_flag.
         bool use_legacy_self_pairing = !ks.uses_tablets();
-        auto target_endpoint = get_view_natural_endpoint(base_ermp, view_ermp, network_topology, base_token, view_token, use_legacy_self_pairing, cf_stats);
+        auto [target_endpoint, extra_target_endpoint] = get_view_natural_endpoint(base_ermp, view_ermp, network_topology, base_token, view_token, use_legacy_self_pairing, cf_stats);
         auto remote_endpoints = view_ermp->get_pending_replicas(view_token);
+        if (extra_target_endpoint) {
+            // The extra endpoint is a view replica whose paired base replica is not ready yet.
+            // It's not pending, so it shouldn't be duplicated in the pending endpoints list.
+            remote_endpoints.push_back(std::move(*extra_target_endpoint));
+        }
```

Comment:
by doing this unconditionally, we're not getting the optimization if the extra view replica happens to be local. but i think it's some significant rework in this area to make that happen. perhaps a FIXME?

Reply:
I seems to me that we should still get the optimization here - setting the local `target_endpoint` happens below based on `remote_endpoints` where we're adding the `extra_target_endpoint`.
A possible concern is if it's duplicated, but I don't think it can be - `remote_endpoints` are initially filled with pending endpoints and the `extra_target_endpoint` is not pending (otherwise we wouldn't try to pair with it)